### ISSUE TITLE
JS enhanced header links

### ISSUE THREAD

**Issue by swantzter** [Libera-Chat/libera-chat.github.io#21]
essentially:

```js
document.querySelectorAll('main :is(h2,h3,h4,h5,h6)')
  .forEach(el => {
    el.innerHTML = `<a href="#${el.id}">${el.innerHTML}</a>`
})
```

and

```css
main :is(h2, h3, h4, h5, h6) a {
  text-decoration: none;
}

main :is(h2, h3, h4, h5, h6) a:hover::before {
  content: '#';
  position: absolute;
  left: 0;
}
```

**Comment by swantzter** [comment]
closed in ef321edcb7c247f0242969e2b8a56dee566fa0f2

**Comment by swantzter** [comment]
Actually, I forgot to account for the header height so the title gets hidden now.

**Comment by swantzter** [comment]
closed again in b9c7faab3c6085fcba5809b66d60e7bf41ed1b18
